File: If-Else/CPP/pr_04_quadratic_equation.cpp
#include<iostream>
#include<cmath>
using namespace std;

int main() {

    float a, b, c, x1, x2, descriminant, real_part, img_part;
    cout<<"Enter the coefficients "<<endl;
    cout<<"Enter the value of a ";
    cin>>a;
    cout<<"Enter the value of b ";
    cin>>b;
    cout<<"Enter the value of c ";
    cin>>c;

    descriminant = b*b - 4*a*c;

    if (descriminant > 0) {
        x1 = (-b + sqrt(descriminant)) / (2*a);
        x2 = (-b - sqrt(descriminant)) / (2*a);

        cout<<"The roots are real and different"<<endl;
        cout<<"x1 = "<<x1<<endl;
        cout<<"x2 = "<<x2<<endl;
    }

    else if (descriminant == 0) {
        cout<<"The roots are real and same"<<endl;
        x1 = -b / (2*a);
        cout<<"x1 = x2 = "<<x1<<endl;
    }

    else {
        real_part = -b/(2*a);
        img_part = sqrt(-descriminant) / (2*a);
        cout<<"The roots are complex and different"<<endl;
        cout<<"x1 = "<<real_part<<" + "<<img_part<<"i"<<endl;
        cout<<"x2 = "<<real_part<<" - "<<img_part<<"i"<<endl;
    }

    return 0;
    
}

File: If-Else/Python/pr_04_quadratic_equation.py
import math
print("Enter the coefficients ")
a = int(input("Enter the value of a "))
b = float(input("Enter the value of b "))
c = float(input("Enter the value of c "))

descriminant = b**2 - 4*a*c

if descriminant > 0:
    x1 = (-b + math.sqrt(descriminant)) / (2*a)
    x2 = (-b - math.sqrt(descriminant)) / (2*a)

    print("The roots are real and different")
    print(f"x1 = {x1}")
    print(f"x2 = {x2}")

elif descriminant == 0:
    x1 = -b / (2*a)
    print("The roots are real and same")
    print(f"x1 = x2 = {x1}")

else:
    real_part = -b / (2*a)
    img_part = math.sqrt(-descriminant) / (2*a)
    print("The roots are complex and different")
    print(f"x1 = {real_part} + {img_part} i")
    print(f"x2 = {real_part} - {img_part} i")
    